modules/deployment-scripts/create-kv-certificate/create-kv-cert.sh

```diff
@@ -1,54 +1,63 @@
+retryMax="${retryMax:-5}"
+certName="${certName:-default-cert}"
+certCommonName="${certCommonName:-default}"
+validity="${validity:-12}"
+akvName="${akvName:-keyvault}"
+issuerName="${issuerName:-}"
+reuseKey="${reuseKey:-true}"
 
-echo "Waiting on Identity RBAC replication ($initialDelay)"
-sleep $initialDelay
+echo "Waiting on Identity RBAC replication (\"$initialDelay\")"
+sleep "$initialDelay"
 
 #Retry loop to catch errors (usually RBAC delays)
 retryLoopCount=0
-until [ $retryLoopCount -ge $retryMax ]
+until [ "$retryLoopCount" -ge "$retryMax" ]
 do
     echo "Creating AKV Cert $certName with CN $certCommonName (attempt $retryLoopCount)..."
 
     if [ -z "$issuerName" ] || [ -z "$issuerProvider" ]; then
         policy=$(az keyvault certificate get-default-policy \
-            | sed -e s/\"validityInMonths\":\ 12/\"validityInMonths\":\ ${validity}/g \
-            | sed -e s/CN=CLIGetDefaultPolicy/CN=${certCommonName}/g )
+            | sed -e s/\"validityInMonths\":\ 12/\"validityInMonths\":\ "${validity}"/g \
+            | sed -e s/CN=CLIGetDefaultPolicy/CN="${certCommonName}"/g )
     else
-      if [ "$issuerProvider" == "DigiCert" ] || [ "$issuerProvider" == "GlobalCert"]; then
+      if [ "$issuerProvider" == "DigiCert" ] || [ "$issuerProvider" == "GlobalCert" ]; then
         az keyvault certificate issuer create \
-          --vault-name $akvName \
-          --issuer-name $issuerName \
-          --provider-name $issuerProvider \
-          --account-id $accountId \
-          --password $issuerPassword \
-          --organizatiion-id $organizationId
+          --vault-name "$akvName" \
+          --issuer-name "$issuerName" \
+          --provider-name "$issuerProvider" \
+          --account-id "$accountId" \
+          --password "$issuerPassword" \
+          --organizatiion-id "$organizationId"
       else
         az keyvault certificate issuer create \
-          --vault-name $akvName \
-          --issuer-name $issuerName \
-          --provider-name $issuerProvider
+          --vault-name "$akvName" \
+          --issuer-name "$issuerName" \
+          --provider-name "$issuerProvider"
       fi
       policy=$(az keyvault certificate get-default-policy \
-        | sed -e s/\"validityInMonths\":\ 12/\"validityInMonths\":\ ${validity}/g \
-        | sed -e s/CN=CLIGetDefaultPolicy/CN=${certCommonName}/g  \
-        | sed -e s/\"name\":\ \"Self\"/\"name\":\ \"${issuerName}\"/g )
+        | sed -e s/\"validityInMonths\":\ 12/\"validityInMonths\":\ "${validity}"/g \
+        | sed -e s/CN=CLIGetDefaultPolicy/CN="${certCommonName}"/g \
+        | sed -e s/\"name\":\ \"Self\"/\"name\":\ \""${issuerName}"\"/g \
+        | sed -e s/\"reuseKey\":\ true/\"reuseKey\":\ "${reuseKey}"/g )
     fi
     az keyvault certificate create \
-        --vault-name $akvName \
-        -n $certName \
+        --vault-name "$akvName" \
+        -n "$certName" \
         -p "$policy" \
         --disabled "$disabled" \
```

Comment:
⚠️ **[shellcheck]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>disabled is referenced but not assigned. [SC2154](https://github.com/koalaman/shellcheck/wiki/SC2154)